internal/provider/resource_integration.go

```diff
@@ -459,55 +460,12 @@ func (r *integrationResource) ValidateConfig(ctx context.Context, req resource.V
 			))
 		}
 	}
-}
-
-func convertMappingsArrayToModel(m []models.ResourceOwnerMapping) []aponoapi.ResourceOwnerMappingTerraform {
-	result := make([]aponoapi.ResourceOwnerMappingTerraform, len(m))
-	for i, rom := range m {
-		result[i] = resourceOwnerMappingToModel(rom)
-	}
-	return result
-}
-
-func resourceOwnerMappingToModel(rom models.ResourceOwnerMapping) aponoapi.ResourceOwnerMappingTerraform {
-	return aponoapi.ResourceOwnerMappingTerraform{
-		TagName:                rom.TagName.ValueString(),
-		AttributeType:          rom.AttributeType.ValueString(),
-		AttributeIntegrationId: *aponoapi.NewNullableString(rom.AttributeIntegrationId.ValueStringPointer()),
-	}
-}
-
-func integrationOwnersToModel(IntegrationOwners []models.IntegrationOwner) []aponoapi.IntegrationOwnerTerraform {
-	owners := make([]aponoapi.IntegrationOwnerTerraform, len(IntegrationOwners))
-	for i, o := range IntegrationOwners {
-		owners[i] = integrationOwnerToModel(o)
-	}
-	return owners
-}
-
-func integrationOwnerToModel(owner models.IntegrationOwner) aponoapi.IntegrationOwnerTerraform {
-	return aponoapi.IntegrationOwnerTerraform{
-		IntegrationId:  *aponoapi.NewNullableString(owner.IntegrationId.ValueStringPointer()),
-		AttributeType:  owner.AttributeType.ValueString(),
-		AttributeValue: convertStringArray(owner.AttributeValue),
-	}
-}
-
-func convertStringArray(a []types.String) []string {
-	result := make([]string, len(a))
-	for i, item := range a {
-		result[i] = item.ValueString()
-	}
-	return result
-}
 
-func toNullableString(s types.String) aponoapi.NullableString {
-	result := aponoapi.NullableString{}
-	if s.IsNull() {
-		result.Unset()
-	} else {
-		vs := s.ValueString()
-		result.Set(&vs)
+	if model.ResourceOwnerMappings != nil && model.IntegrationOwners == nil {
+		resp.Diagnostics.AddAttributeError(
+			path.Root("resource_owner_mappings"),
+			"Invalid Configuration",
+			"`resource_owner_mappings` cannot be set unless `integration_owners` is also set.",
```

Comment:
you can define integration owner without resource owners? if so this validation is not good, if not lets change the message to be more like "both field must be set together"